src/view/document.js

```diff
@@ -300,6 +301,21 @@ export default class Document {
 	}
 
 	/**
+	 * Scrolls the page viewport and {@link #domRoots} with their ancestors to reveal the
+	 * caret, if not already visible to the user.
+	 */
+	scrollToTheSelection() {
+		const range = this.selection.getFirstRange();
+
+		if ( range ) {
+			scrollViewportToShowTarget( {
+				target: this.domConverter.viewRangeToDom( range ),
```

Comment:
I haven't noticed this before, but the `target` option isn't the most flexible one: https://github.com/ckeditor/ckeditor5-utils/issues/176.